### PR TITLE
Fix Required Maps after UI Review v2

### PR DESCRIPTION
**Pull request by nazar-kutz**

### CODE REVIEW

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 1]
```diff
@@ -93,24 +94,24 @@ public boolean checkIfThereAreMissingMaps(RoutingContext ctx, LatLon start, List
 		if (end != null) {
 			addPoint(ctx, knownMaps, pointsToCheck, end);
 		}
-		Set<String> usedMaps = new TreeSet<String>();
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:96]: Much easier to use LinkedHashSet

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 2]
```diff
@@ -23,10 +22,12 @@
 import net.osmand.map.OsmandRegions;
 import net.osmand.map.WorldRegion;
 import net.osmand.util.Algorithms;
+import net.osmand.util.CollectionUtils;
 import net.osmand.util.MapUtils;
 
 public class MissingMapsCalculator {
-	protected static final Log log = PlatformUtil.getLog(MissingMapsCalculator.class);
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:29]: Extra task to remove static:
```
public static void initMissingMapsCalculator(OsmandRegions osmandRegions) {
		if (missingMapsCalculator == null) {
			missingMapsCalculator = new MissingMapsCalculator(osmandRegions);
		}
	}
```
Idea: 
Add static OsmandRegions to PlatformUtil class, MissingMapsCalculator should be always initiated as used no static instance. 	
```
public MissingMapsCalculator() throws IOException {
```
Should be deleted

**Comment by vshcherb** [comment]
Task 4.8

File: OsmAnd-java/src/main/java/net/osmand/PlatformResources.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package net.osmand;
+
+import net.osmand.map.OsmandRegions;
+
+import java.io.IOException;
+
+public class PlatformResources {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/PlatformResources.java:7]: 1. PlatformUtil 
2. Broken backend 
3. Not reusing Android object

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 4]
```diff
@@ -158,12 +161,14 @@ public boolean checkIfThereAreMissingMaps(RoutingContext ctx, LatLon start, List
 		if (mapsToDownload.isEmpty() && mapsToUpdate.isEmpty()) {
 			return false;
 		}
-		ctx.calculationProgress.requestMapsToUpdate = true;
-		ctx.calculationProgress.missingMaps = convert(mapsToDownload);
-		ctx.calculationProgress.mapsToUpdate = convert(mapsToUpdate);
-		ctx.calculationProgress.potentiallyUsedMaps = convert(usedMaps);
+		RouteCalculationProgress progress = ctx.calculationProgress;
+		progress.requestMapsToUpdate = true;
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:165]: should create MissingMapsCalculationResult to not overuse Progress

**Comment by vshcherb** [comment]
- Is fixed order of maps part of this request or another? 
- PlatformUtil will be separate request ?

**Comment by nazar-kutz** [comment]
This request contains both:
1. Fix maps order
2. PlatformUtil fix

**Review by vshcherb — APPROVED** [review]

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 5]
```diff
@@ -93,24 +95,25 @@ public boolean checkIfThereAreMissingMaps(RoutingContext ctx, LatLon start, List
...
+		Set<String> usedMaps = new LinkedHashSet<>();
+		Set<String> mapsToDownload = new LinkedHashSet<>();
+		Set<String> missingMaps = new LinkedHashSet<>();
+		Set<String> mapsToUpdate = new LinkedHashSet<>();
 		Set<Long> presentTimestamps = null;
 		for (Point p : pointsToCheck) {
 			if (p.hhEditions == null) {
 				if (p.regions.size() > 0) {
-					mapsToDownload.add(p.regions.get(0));
+					CollectionUtils.addIfNotContains(p.regions.get(0), missingMaps, mapsToDownload);
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:106]: why is it needed for set? Just add is good enough

Comment (nazar-kutz) [reply OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java]: Fixed in https://github.com/osmandapp/OsmAnd/pull/19695/commits/09d3b74bb496b4138303a2b3abd03185fcfc9b82

**Pull request merged**